### ISSUE TITLE
Improperly formatted code-block triggers false-positives of MD031

### ISSUE THREAD

**Issue by NVedsted** [markdownlint/markdownlint#524]
## Description
Invalid, single-line fenced code block triggers false-positives MD031 for the rest of the file.

### Environment
**MDL Version:** 0.13.0

### Expected Behavior
Given the below example, I would expect no errors for the following correctly formatted code blocks. It would be nice to get an error for the improperly formatted code block on line 1, but that is not relevant for this bug necessarily. (EDIT: I suppose it is actually a valid in-line code block which should not error.)

### Actual Behavior
The below example, yields the following linting errors:

```
/code/README.md:3: MD031 Fenced code blocks should be surrounded by blank lines
/code/README.md:5: MD031 Fenced code blocks should be surrounded by blank lines
```

These are clearly false-positives.

## Replication Case
````markdown
```test```

```text
test
```

````
